src/phone-number-field-lite.tsx

```diff
@@ -0,0 +1,186 @@
+        } else if (digits.length > 2 && digits.length <= 11 && digits[2] <= '5') {
+            return `(${digits.slice(0, 2)}) ${digits.slice(2, 6)}-${digits.slice(6)}`.replace(/\D+$/, '');
+        }
+    } else if (regionCode === 'DE') {
+        // https://en.wikipedia.org/wiki/Telephone_numbers_in_Germany
+        // Only formatting mobile numbers, landline numbers have a lot of variations:
+        // https://en.wikipedia.org/wiki/Telephone_numbers_in_Germany#/media/File:Karte_Telefonvorwahlen_Deutschland.png
+        if (digits.length >= 4 && digits.match(/^(015|016|017)/)) {
+            if (digits.length <= 12 && digits.startsWith('015')) {
+                return `${digits.slice(0, 5)} ${digits.slice(5)}`.trim();
+            } else {
+                return `${digits.slice(0, 4)} ${digits.slice(4)}`.trim();
+            }
+        }
+    } else if (regionCode === 'GB') {
+        // https://en.wikipedia.org/wiki/Telephone_numbers_in_the_United_Kingdom#Mobile_telephones
+        // Like in DE, only mobile numbers are formatted
+        // Example mobile: 07xxx xxxxxx
+        if (digits.length <= 11 && digits.startsWith('07')) {
+            return `${digits.slice(0, 5)} ${digits.slice(5)}`.trim();
+        }
+    }
+    return digits;
+};
+
+type InputProps = Omit<React.InputHTMLAttributes<HTMLInputElement>, 'value' | 'onInput'> & {
+    inputRef?: React.Ref<HTMLInputElement>;
+    value?: string;
+    defaultValue?: string;
+    onInput?: (event: React.FormEvent<HTMLInputElement>) => void;
+    prefix?: string;
+    format?: (number: string) => string;
+};
+
+const PhoneInput = ({
+    inputRef,
+    value,
+    defaultValue,
+    onChange,
+    prefix,
+    format: formatFromProps,
+    ...other
+}: InputProps) => {
+    const [selfValue, setSelfValue] = React.useState(defaultValue || '');
+    const ref = React.useRef<HTMLInputElement | null>(null);
+    const {i18n} = useTheme();
+
+    const regionCode = i18n.phoneNumberFormattingRegionCode;
+    const isControlledByParent = typeof value !== 'undefined';
+    const controlledValue = (isControlledByParent ? value : selfValue) as string;
+
+    const handleChangeValue = React.useCallback(
+        (newFormattedValue: string) => {
+            if (!isControlledByParent) {
+                setSelfValue(newFormattedValue);
+            }
+            if (ref.current) {
+                onChange?.(createChangeEvent(ref.current, newFormattedValue));
+            }
+        },
+        [isControlledByParent, onChange]
+    );
+
+    const format = React.useCallback(
+        (number: string): string => {
+            if (formatFromProps) {
+                return formatFromProps(number);
+            }
+            return formatPhoneLite(regionCode, number);
+        },
+        [formatFromProps, regionCode]
+    );
+
+    const rifm = useRifm({
+        format,
+        value: controlledValue,
+        accept: /[\d\+]+/g,
```

Comment:
@marcoskolodny the issues we saw in the custom formatter example and the `+` char were caused because it wasn't added in the `accept` regex

Reply:
Good catch!